src/Exceptionless.Insulation/Bootstrapper.cs

```diff
@@ -26,162 +26,188 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using Serilog.Sinks.Exceptionless;
 using StackExchange.Redis;
 
 namespace Exceptionless.Insulation {
     public class Bootstrapper {
-        public static void RegisterServices(IServiceCollection container, bool runMaintenanceTasks) {
-            if (!String.IsNullOrEmpty(Settings.Current.ExceptionlessApiKey) && !String.IsNullOrEmpty(Settings.Current.ExceptionlessServerUrl)) {
+        public static void RegisterServices(IServiceProvider serviceProvider, IServiceCollection services, AppOptions appOptions, bool runMaintenanceTasks) {
```

Comment:
I don't love that we are passing in IServiceProvider here instead of building it in here where we are mostly using it. I see why you did it because of the job bootstrapper needing it, but I'm wondering if we could refactor that some so that logic moves in here.